src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ADD_ONE_ELEMENT = 1;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[])new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < elements.length) {
+            elements[size++] = value;
+        } else {
+            resizeIfNeeded(ADD_ONE_ELEMENT);
+            elements[size++] = value;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        verifyIndexForAdd(index);
+        resizeIfNeeded(ADD_ONE_ELEMENT);
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        resizeIfNeeded(list.size());
+        int indexOfLastElement = size;
+        for (int i = 0; i < list.size(); i++) {
+            elements[indexOfLastElement++] = list.get(i);
+            size++;
+        }
```

Comment:
Actually, you can simplify it. I'd recommend just iterating (for-each) through the list and using your add() method to add new element

Reply:
I can`t use for each for Generics foreach not applicable to type 'core.basesyntax.List<T>'